### ISSUE TITLE
PHP gRPC Client Failed to Connect to All Address, Invalid Argument

### ISSUE THREAD

**Issue by rigasdannbs** [grpc/grpc#34802]
### What version of gRPC and what language are you using?
programming language: php-8.1
composer.json:  grpc/grpc 1.42 

### What operating system (Linux, Windows,...) and version?
Ubuntu

### What runtime / compiler are you using (e.g. python version or version of gcc)
php-fpm8.1

### What did you do?
Connecting api via grpc protocol, got intermittent error. <br>

The error response:
```
{
    "message": "failed to connect to all addresses; last error: UNKNOWN: ipv4:172.16.153.76:10000: Invalid argument"
}
```

references image from the intermittent error: [link](https://drive.google.com/file/d/1sItP-pi07f9hXUF9Kg8J_IA3SungFZh9/view?usp=sharing)

### What did you expect to see?
all of the request should be status code: 200, OK

### What did you see instead?

and trace the logs from TROUBLESHOOT.md in grpc documentation. here is the result of the TRACE grpc

```
I1698289980.715278620 handshaker.cc:100]                     handshake_manager 0x7f8904004060: error=UNKNOWN:Invalid argument {target_address:"ipv4:163.47.9.141:10000", syscall:"connect", os_error:"Invalid argument", errno:22, created_time:"2023-10-26T03:13:00.713770253+00:00", file_line:365, file:"/tmp/pear/temp/grpc/src/core/lib/iomgr/tcp_client_posix.cc"} shutdown=0 index=1, args={endpoint=(nil), args={}, read_buffer=(nil) (length=0), exit_early=0}
I1698289980.715310554 handshaker.cc:132]                     handshake_manager 0x7f8904004060: handshaking complete -- scheduling on_handshake_done with error=UNKNOWN:Invalid argument {target_address:"ipv4:163.47.9.141:10000", syscall:"connect", os_error:"Invalid argument", errno:22, created_time:"2023-10-26T03:13:00.713770253+00:00", file_line:365, file:"/tmp/pear/temp/grpc/src/core/lib/iomgr/tcp_client_posix.cc"}
```

and others line in the log from TRACE
```
D1698289980.716594218 exec_ctx.cc:36]                        running closure 0x7f8904007f40: created [/tmp/pear/temp/grpc/src/core/lib/iomgr/closure.h:167]: scheduled [/tmp/pear/temp/grpc/src/core/ext/filters/client_channel/client_channel.cc:3201]
I1698289980.716615513 client_channel.cc:2639]                chand=0x55efd709b400 lb_call=0x7f88f4003840: grabbing LB mutex to get picker
I1698289980.716635572 dual_ref_counted.h:258]                SubchannelPicker:0x7f8904003c00 ref 1 -> 2; (weak_refs=0)
I1698289980.716654571 client_channel.cc:2649]                chand=0x55efd709b400 lb_call=0x7f88f4003840: performing pick with picker=0x7f8904003c00
I1698289980.716677037 client_channel.cc:2749]                chand=0x55efd709b400 lb_call=0x7f88f4003840: LB pick failed: UNAVAILABLE: failed to connect to all addresses; last error: UNKNOWN: ipv4:163.47.9.141:10000: Invalid argument
I1698289980.716719469 client_channel.cc:2678]                chand=0x55efd709b400 lb_call=0x7f88f4003840: failed to pick subchannel: error=UNKNOWN:failed to connect to all addresses; last error: UNKNOWN: ipv4:163.47.9.141:10000: Invalid argument {file:"/tmp/pear/temp/grpc/src/core/lib/transport/error_utils.cc", file_line:153, created_time:"2023-10-26T03:13:00.716701205+00:00", grpc_status:14}
```

### Anything else we should know about your project / environment?
Running in the top of the Docker Container by using this image php:8.1-fpm, and deployed it on the kubernetes environment

**Comment by alto-php** [comment]
Can you share the code you use to setup the client and connect to your server?

**Comment by benjaminp** [comment]
This may be fixed by #35064.

**Comment by rigasdannbs** [comment]
Btw our solution using pecl grpc version 1.59.1
